### DISCUSSION TITLE
Bulk delete AddressObject and AddressGroup entries

### DISCUSSION THREAD

**Discussion by da110us** [PaloAltoNetworks/pan-os-python#551] [General]
I'm struggling with the SDK: I want to bulk delete  AddressObject and AddressGroup entries.
The way it is coded in the example "Optimize with Bulk Operations" for Create does not work for delete_similar().
The API reference also does not provide much more  information for delete_similar(). Maybe I'm on the complete wrong track ;-)

Can somebody explain to me how to delete all AddressObject in bulk (so with delete_similar() )? Or maybe somebody has some sample code how to bulk delete these objects?

**Comment by nikolay-matveev-kkr** [comment]
It’s not you or the SDK, it’s PAN-OS, unfortunately.

The bulk deletion feature has been disabled recently (because of some underlying issues).
You can still use create_similar() for address object but not delete_similar() which is a shame.

I think this really affects many customers who use automation (it does affect my company) but nobody in Palo cared to notify customers before the feature was disabled, nobody explained why this was done, and nobody promissed to fix this either -  to the best of my knowledge.

I really hope that at least one responsible person in the PAN-OS product development team folows this Python SDK repo and can shed some light on the problem.

From: da110us ***@***.***>
Sent: Monday, March 11, 2024 11:55 AM
To: PaloAltoNetworks/pan-os-python ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [PaloAltoNetworks/pan-os-python] Bulk delete AddressObject and AddressGroup entries (Discussion #551)

I'm struggling with the SDK: I want to bulk delete AddressObject and AddressGroup entries. The way it is coded in the example "Optimize with Bulk Operations" for Create does not work for delete_similar(). The API reference also does not provide


I'm struggling with the SDK: I want to bulk delete AddressObject and AddressGroup entries.
The way it is coded in the example "Optimize with Bulk Operations" for Create does not work for delete_similar().
The API reference also does not provide much more information for delete_similar(). Maybe I'm on the complete wrong track ;-)

Can somebody explain to me how to delete all AddressObject in bulk (so with delete_similar() )? Or maybe somebody has some sample code how to bulk delete these objects?

—
Reply to this email directly, view it on GitHub<https://urldefense.com/v3/__https:/github.com/PaloAltoNetworks/pan-os-python/discussions/551__;!!HWWtag!xAHeA6dixyfe8nTUCUnPACH_1ECLyd84iEGkuXnvCHg_ZITgzQ0qnN88ryM7b8r_CNQNI7e6wWoF8nhnpK0GByBGEyFx$>, or unsubscribe<https://urldefense.com/v3/__https:/github.com/notifications/unsubscribe-auth/AYBODG2PPLT7AQ425QY5W6LYXWLRZAVCNFSM6AAAAABEQE3NSGVHI2DSMVQWIX3LMV43ERDJONRXK43TNFXW4OZWGM2TOMBUGU__;!!HWWtag!xAHeA6dixyfe8nTUCUnPACH_1ECLyd84iEGkuXnvCHg_ZITgzQ0qnN88ryM7b8r_CNQNI7e6wWoF8nhnpK0GB0RdseJx$>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.******@***.***>>

===============================================================================
Please refer to http://www.kkr.com/email-disclaimer for important disclosures regarding this electronic communication.
KKR uses your personal information in accordance with its privacy notices, which can be found on http://www.kkr.com.
===============================================================================

**Reply by da110us** [reply]
Many, many thanks for this reply! Saves me a lot of wasted time searching for the impossible,

I think I have an issues then, because I need to delete 90K of objects. Wondering how Expedition will solve this. You would think it will take the same time to delete unused objects.....

**Reply by da110us** [reply]
PS: Is it known which version this function is removed? I'm testing on 10.2, but the production firewalls still run on 9.1

**Reply by nikolay-matveev-kkr** [reply]
It definitely worked for me in 9.1 until a certain minor version (cannot recall which one exactly it was) but then stoped, ands shortly thereafter we upgraded to 10.1 & 10.2 where the feature did not work either.

For 90K objects I'd definitely go for something like **pan-os-php** or **Expedition** with both of them working on offline configuration file. I think any API would be a bad idea for that number of objects :)
